HUD/hud_scraper.py

```diff
@@ -0,0 +1,177 @@
+        ],
+        ignore_index=True
+    ).reset_index(drop=True)
+    df['source'] = ['HUD'] * len(df)
+    df['service_type'] = ['SHELTER'] * len(df)
+    service_summaries = []
+    for i in tqdm(range(len(df))):
+        if df.loc[i, 'Project Type'] == 'ES':
+            if df.loc[i, 'Target Population'] is not None:
+                service_summaries.append('Domestic Violence Shelter')
+            else:
+                service_summaries.append('Emergency Shelter')
+        else:
+            service_summaries.append('Transitional Housing')
+    df['service_summary'] = service_summaries
+    return df
+
+
+def scrape_updated_date():
+    url = 'https://www.hudexchange.info/resource/3031/pit-and-hic-data-since-2007/'
+    resp = requests.get(url).text
+    update_statement = re.search(
+        r'Date Published: (\w{3,8} \d{4})', resp
+    )
+    scraped_date = datetime.strptime(
+        update_statement.group(1), "%B %Y"
+    ).date()
+    return scraped_date
+
+
+def check_site_for_new_date(existing_date):
+    """Check HUD web page with data files to see if the most
+    recently updated date is different than the date in MongoDB
+
+    Args:
+        existing_date (datetime): the date that the IRS last updated,
+        according to the data-sources collection
+
+    Returns:
+        bool: whether or not the dates are different
+    """
+    scraped_date = scrape_updated_date()
+    return scraped_date > existing_date
+
+
+def prevent_HUD_duplicates(pID, client, collection):
+    coll = client[collection]
+    dupe = coll.find_one(
+        {'Project ID': pID}
+    )
+    return dupe
+
+
+def purge_HUD_duplicates(df, client, collection, dupe_collection):
+    found_duplicates = []
+    for i in range(len(df)):
+        pID = int(df.loc[i, 'Project ID'])
+        if prevent_HUD_duplicates(pID, client, collection):
+            found_duplicates.append(i)
+    duplicate_df = df.loc[found_duplicates].reset_index(drop=True)
+    print('inserting tmpHUD dupes into the dupe collection')
+    insert_services(duplicate_df.to_dict('records'), client, dupe_collection)
+    df = df.drop(found_duplicates).reset_index(drop=True)
+    return df
+
+
+def main(client, check_collection, dump_collection, dupe_collection):
+    scraped_update_date = scrape_updated_date()
+    print(str(scraped_update_date))
+    try:
+        stored_update_date = client['data-sources'].find_one(
+            {"name": "hud_pit_hic_data"}
+        )['last_updated']
+        stored_update_date = datetime.strptime(
+            str(stored_update_date), '%Y-%m-%d %H:%M:%S'
+        ).date()
+        print(stored_update_date)
+        print(check_site_for_new_date(stored_update_date))
+        if not check_site_for_new_date(stored_update_date):
+            print('No new update detected. Exiting script...')
+            return
+    except KeyError:
+        print('Key Error')
+        pass
+    print('updating scraped update date in data-sources collection')
+    try:
```

Comment:
Should we do it after the scraping is complete? I mean what if scrapping fails but we updated the last updated date for it.